From 6119e8e25d7d38dcc05f8d1a7efc16ce27ef2f43 Mon Sep 17 00:00:00 2001
From: Vincent Untz <vuntz@suse.com>
Date: Tue, 14 Nov 2017 17:53:32 +0100
Subject: [PATCH] Catch socket.timeout when doing heartbeat_check

heartbeat_check in kombu.connection is not reraising exceptions as
exceptions.OperationalError, and the socket timeout during the heartbeat
check is really an issue seen in the field when a node goes down; the
heartbeat thread just tries again and again to deal with it, without
success.

Change-Id: I26dbdb18a7e64946db2cba676764ff2d428c7897
Closes-Bug: #1657444
(cherry picked from commit 8bfc3637a25a29583b1e0625c78bf159ac878259)
---
 oslo_messaging/_drivers/impl_rabbit.py | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/oslo_messaging/_drivers/impl_rabbit.py b/oslo_messaging/_drivers/impl_rabbit.py
index b517240..a61d5b8 100644
--- a/oslo_messaging/_drivers/impl_rabbit.py
+++ b/oslo_messaging/_drivers/impl_rabbit.py
@@ -961,7 +961,8 @@ class Connection(object):
 
                 recoverable_errors = (
                     self.connection.recoverable_channel_errors +
-                    self.connection.recoverable_connection_errors)
+                    self.connection.recoverable_connection_errors +
+                    (socket.timeout,))
 
                 try:
                     try:
-- 
2.15.0

